detectors/resources/src/test/java/com/google/cloud/opentelemetry/detectors/GCEResourceTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Map;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GCEResourceTest {
+  @Rule public final WireMockRule wireMockRule = new WireMockRule(8089);
+  private final GCPMetadataConfig metadataConfig = new GCPMetadataConfig("http://localhost:8089/");
+  private final GCEResource testResource = new GCEResource(metadataConfig);
+
+  // Helper method to help stub endpoints
+  private void stubEndpoint(String endpointPath, String responseBody) {
+    stubFor(
+        get(urlEqualTo(endpointPath))
+            .willReturn(
+                aResponse()
+                    .withStatus(200)
+                    .withHeader("Content-Type", "text/plain")
```

Comment:
NIt:  AFAIK there's a header we can add here to confirm something came from google cloud metadata.

Can you add: "Metadata-Flavor: Google" here and then require that in your HTTP calling code?

Reply:
It's actually both a request + response header.

(a) We should use it in the request header
(b) For now, I think we use it in the response header and we can open a bug to follow up and make sure this mechanism remains.

Example curl output from a GCE vm:

```
$ curl -v "http://metadata.google.internal/computeMetadata/v1/instance/tags" -H "Metadata-Flavor: Google"
...
< HTTP/1.1 200 OK
< Metadata-Flavor: Google
< Content-Type: application/json
< ETag: {redacted}
< Date: Fri, 05 Feb 2021 14:03:45 GMT
< Server: Metadata Server for VM
< Content-Length: 2
< X-XSS-Protection: 0
< X-Frame-Options: SAMEORIGIN
```